bigquery/list_datasets_and_projects.js

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Command-line application to list all projects and datasets in BigQuery.
+ *
+ * This sample is used on this page:
+ *
+ *   https://cloud.google.com/bigquery/docs/managing_jobs_datasets_projects
+ */
+
+'use strict';
+
+// [START auth]
+// By default, gcloud will authenticate using the service account file specified
+// by the GOOGLE_APPLICATION_CREDENTIALS environment variable and use the
+// project specified by the GCLOUD_PROJECT environment variable. See
+// https://googlecloudplatform.github.io/gcloud-node/#/docs/guides/authentication
+var gcloud = require('gcloud');
+// [END auth]
+
+// [START list_tables]
+/**
+ * Retrieve all datasets for the specified project.
+ *
+ * @param {string} projectId The project to get datasets from.
+ * @param {Function} callback Callback function.
+ */
+function listDatasets (projectId, callback) {
+  if (!projectId) {
+    throw new Error('projectId is required');
+  }
+
+  var bigquery = gcloud.bigquery({
+    projectId: projectId
+  });
+
+  bigquery.getDatasets(function (err, datasets, nextQuery, apiResponse) {
+    // Quit on error
+    if (err) {
+      return callback(err);
+    }
+
+    // Pagination
```

Comment:
Never mind - happened across [this page](https://cloud-dot-devsite.googleplex.com/bigquery/docs/data).